addons/account/models/company.py

```diff
@@ -73,27 +74,72 @@ def _verify_fiscalyear_last_day(self, company_id, last_day, last_month):
         return last_day > last_day_of_month and last_day_of_month or last_day
 
     @api.multi
-    def compute_fiscalyear_dates(self, date):
-        """ Computes the start and end dates of the fiscalyear where the given 'date' belongs to
-            @param date: a datetime object
-            @returns: a dictionary with date_from and date_to
-        """
-        self = self[0]
-        last_month = self.fiscalyear_last_month
-        last_day = self.fiscalyear_last_day
-        if (date.month < last_month or (date.month == last_month and date.day <= last_day)):
-            date = date.replace(month=last_month, day=last_day)
-        else:
-            if last_month == 2 and last_day == 29 and (date.year + 1) % 4 != 0:
-                date = date.replace(month=last_month, day=28, year=date.year + 1)
-            else:
-                date = date.replace(month=last_month, day=last_day, year=date.year + 1)
-        date_to = date
-        date_from = date + timedelta(days=1)
-        if date_from.month == 2 and date_from.day == 29:
-            date_from = date_from.replace(day=28, year=date_from.year - 1)
+    def compute_fiscalyear_dates(self, current_date):
+        '''Computes the start and end dates of the fiscal year where the given 'date' belongs to.
+
+        :param current_date: A datetime.date/datetime.datetime object.
+        :return: A dictionary containing:
+            * date_from
+            * date_to
+            * [Optionally] record: The fiscal year record.
+        '''
```

Comment:
that function is not really readable....

```python
def compute_fiscalyear_dates(self, current_date):
    #search fiscal year with current_date between date_start and date_end
    fiscalyear = self.env['account.fiscal.year'].search([
               ('company_id', '=', self.id),
              ('date_start', '<=', current_date),
              ('date_end, '>=', current_date),
￼    ], order='date_start DESC', limit=1)
    if fiscal_year :
        return  {'date_from': fiscalyear.date_start, 'date_to': fiscal_year.date_end 'record': fiscalyear}
    #otherwise compute date_from and date_to based on company's settings
    ...
    #modify lower/upper bounds in case of fiscal year overlapping with computed dates
    ...
    return {}
```
also the subfunction resolve_february() is badly named, as it actually aims to compute the date_to and not only deal with leap years or february month